sputnikdao-factory2/src/factory_manager.rs

```diff
@@ -0,0 +1,167 @@
+            sys::storage_write(u64::MAX as _, 1 as _, u64::MAX as _, 0 as _, 2);
+            // Load register 1 into blob_hash.
+            let blob_hash = [0u8; 32];
+            sys::read_register(1, blob_hash.as_ptr() as _);
+            // Return from function value of register 1.
+            let blob_hash_str = serde_json::to_string(&Base58CryptoHash::from(blob_hash))
+                .unwrap()
+                .into_bytes();
+            sys::value_return(blob_hash_str.len() as _, blob_hash_str.as_ptr() as _);
+        }
+    }
+
+    /// Delete code from the contract.
+    pub fn delete_contract(&self, code_hash: Base58CryptoHash) {
+        let code_hash: CryptoHash = code_hash.into();
+        env::storage_remove(&code_hash);
+    }
+
+    /// Get code for given hash.
+    pub fn get_code(&self, code_hash: Base58CryptoHash) {
+        let code_hash: CryptoHash = code_hash.into();
+        unsafe {
+            // Check that such contract exists.
+            assert_eq!(
+                sys::storage_has_key(code_hash.len() as _, code_hash.as_ptr() as _),
+                1,
+                "Contract doesn't exist"
+            );
+            // Load the hash from storage.
+            sys::storage_read(code_hash.len() as _, code_hash.as_ptr() as _, 0);
+            // Return as value.
+            sys::value_return(u64::MAX as _, 0 as _);
+        }
+    }
+
+    /// Forces update on the given contract.
+    /// Contract must support update by factory for this via permission check.
+    pub fn update_contract(
+        &self,
+        account_id: AccountId,
+        code_hash: Base58CryptoHash,
+        method_name: &str,
+    ) {
+        let code_hash: CryptoHash = code_hash.into();
+        let account_id = account_id.as_bytes().to_vec();
+        unsafe {
+            // Check that such contract exists.
+            assert_eq!(
+                sys::storage_has_key(code_hash.len() as _, code_hash.as_ptr() as _),
+                1,
+                "Contract doesn't exist"
+            );
+            // Load the hash from storage.
+            sys::storage_read(code_hash.len() as _, code_hash.as_ptr() as _, 0);
+            // Create a promise toward given account.
+            let promise_id =
+                sys::promise_batch_create(account_id.len() as _, account_id.as_ptr() as _);
+            // Call `update` method, which should also handle migrations.
+            sys::promise_batch_action_function_call(
+                promise_id,
+                method_name.len() as _,
+                method_name.as_ptr() as _,
+                u64::MAX as _,
+                0,
+                &NO_DEPOSIT as *const u128 as _,
+                (env::prepaid_gas() - env::used_gas() - GAS_UPDATE_LEFTOVER).0,
+            );
+            sys::promise_return(promise_id);
+        }
+    }
+
+    /// Create given contract with args and callback factory.
+    pub fn create_contract(
```

Comment:
`create_new_dao_contract`?

Reply:
nope, this is generic library - will move to near sdk later